sdks/builds/ios.mk

```diff
@@ -39,6 +39,8 @@ _ios-$(1)_CXX=$$(CCACHE) $$(PLATFORM_BIN)/clang++
 
 _ios-$(1)_AC_VARS= \
 	ac_cv_c_bigendian=no \
+	ac_cv_func_fstatat=no \
+	ac_cv_func_readlinkat=no \
```

Comment:
Can we move this  stuff to configure.ac?

Reply:
IIRC no, because configure checks pass. The problem is that configure verifies that the function exists in the current SDK, but that doesn't mean the function existed in the earliest OS version we support, and at link time we end up with errors about functions that must be weakly linked (note that weak linking isn't the solution, because then the error would just happen at runtime instead, on older Ones - the solution is to avoid the function altogether).

Reply:
You can still hardcode it in configure.ac. Or ifdef it in the code. For people that run configure and not SDK build.

Reply:
I just copied how we do it for simulator builds: https://github.com/mono/mono/blob/3084cf9922d23ef731967440c92e97562e555443/sdks/builds/ios.mk#L185-L186

I'm fine with changing the logic, but I think it should be done everywhere, which makes it out of scope for this PR.

Reply:
@rolfbjarne that’s good as-is